Source/JavaScriptCore/wasm/WasmSlowPaths.cpp

```diff
@@ -506,7 +506,7 @@ inline SlowPathReturnType doWasmCallIndirect(CallFrame* callFrame, Wasm::Instanc
         WASM_THROW(Wasm::ExceptionType::NullTableEntry);
 
     const auto& callSignature = CALLEE()->signature(typeIndex);
-    if (callSignature != Wasm::TypeInformation::getFunctionSignature(function.typeIndex))
+    if (callSignature.index() != function.typeIndex)
```

Comment:
The original check should never find that the modules aren't the same, right?

Reply:
Did you mean the function signatures aren't the same? In that case, no, the check could fail because a Wasm table could be initialized with type `funcref` and the function type matching wouldn't be enforced statically.

(but you're right that it cannot fail if typed function references are used and the table is declared with a concrete function type)